### PR TITLE
feat(plugin): Additional setting for TypingIndicator

### PR DESCRIPTION
**Pull request by Andrew6rant**
This PR allows the user to choose whether or not the plugin's typing indicator appears on the channel sidebar of the currently selected channel.

The default value is `TRUE`, as to maintain the same functionality as previous versions of the plugin.

TypingIndicator

### CODE REVIEW

File: src/plugins/typingIndicator/index.tsx [hunk 1]
```diff
@@ -19,9 +19,10 @@
 import { definePluginSettings, Settings } from "@api/Settings";
 import ErrorBoundary from "@components/ErrorBoundary";
 import { Devs } from "@utils/constants";
+import { getCurrentChannel } from "@utils/discord";
```
Comment (Nuckyz) [inline src/plugins/typingIndicator/index.tsx:22]:
```suggestion
```

File: src/plugins/typingIndicator/index.tsx [hunk 2]
```diff
@@ -47,14 +48,18 @@ function TypingIndicator({ channelId }: { channelId: string; }) {
             return oldKeys.length === currentKeys.length && currentKeys.every(key => old[key] != null);
         }
     );
-
+    const { id: currentChannelId, guild_id: currentGuildId } = useStateFromStores([SelectedChannelStore], () => getCurrentChannel());
```
Comment (Nuckyz) [inline src/plugins/typingIndicator/index.tsx:51]:
```suggestion
    const id: string = useStateFromStores([SelectedChannelStore], () =>SelectedChannelStore.getChannelId());
```

File: src/plugins/typingIndicator/index.tsx [hunk 3]
```diff
@@ -47,14 +48,18 @@ function TypingIndicator({ channelId }: { channelId: string; }) {
...
     const guildId = ChannelStore.getChannel(channelId).guild_id;
 
     if (!settings.store.includeMutedChannels) {
         const isChannelMuted = UserGuildSettingsStore.isChannelMuted(guildId, channelId);
         if (isChannelMuted) return null;
     }
 
+    if (!settings.store.includeCurrentChannel) {
+        if (currentChannelId == channelId) return null;
```
Comment (Nuckyz) [inline src/plugins/typingIndicator/index.tsx:60]:
```suggestion
        if (currentChannelId === channelId) return null;
```

**Review by Nuckyz — CHANGES_REQUESTED** [review]

File: src/plugins/typingIndicator/index.tsx [hunk 4]
```diff
@@ -47,14 +47,18 @@ function TypingIndicator({ channelId }: { channelId: string; }) {
             return oldKeys.length === currentKeys.length && currentKeys.every(key => old[key] != null);
         }
     );
-
+    const id: string = useStateFromStores([SelectedChannelStore], () => SelectedChannelStore.getChannelId());
```
Comment (Nuckyz) [inline src/plugins/typingIndicator/index.tsx:50]:
```suggestion
    const currentChannelId: string = useStateFromStores([SelectedChannelStore], () => SelectedChannelStore.getChannelId());
```

I did an oopsie my bad

Comment (Andrew6rant) [reply src/plugins/typingIndicator/index.tsx:50]: Nah, that one was on me. I saw a suggested change from a contributor to this project and went "LGTM 👍" and committed without actually building the changes to review it

**Review by Nuckyz — APPROVED** [review]

**Pull request merged**